src/github-handler/comment-handler/remote-patch-ranges-handler.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Octokit} from '@octokit/rest';
+import {FileRanges, PatchText, Range, RepoDomain} from '../../types';
+import {getGitHubPatchRanges} from './github-patch-format-handler';
+import {logger} from '../../logger';
+
+/**
+ * For a pull request, get each remote file's patch text asynchronously
+ * Also get the list of files whose patch data could not be returned
+ * @param {Octokit} octokit the authenticated octokit instance
+ * @param {RepoDomain} remote the remote repository domain information
+ * @param {number} pullNumber the pull request number
+ * @param {number} pageSize the number of results to return per page
+ * @returns {Promise<Object<PatchText, string[]>>} the stringified patch data for each file and the list of files whose patch data could not be resolved
+ */
+export async function getCurrentPulLRequestPatches(
+  octokit: Octokit,
+  remote: RepoDomain,
+  pullNumber: number,
+  pageSize: number
+): Promise<{patches: PatchText; filesMissingPatch: string[]}> {
+  // TODO support pagination
+  const filesMissingPatch: string[] = [];
+  const files = (
+    await octokit.pulls.listFiles({
+      owner: remote.owner,
+      repo: remote.repo,
+      pull_number: pullNumber,
+      per_page: pageSize,
+    })
+  ).data;
+  const patches: PatchText = new Map<string, string>();
+  if (files.length === 0) {
+    logger.error(
+      `0 file results have returned from list files query for Pull Request #${pullNumber}. Cannot make suggestions on an empty Pull Request`
+    );
+    throw Error('Empty Pull Request');
+  }
+  files.forEach(file => {
+    if (file.patch === undefined) {
+      // files that are too large do not return the patch text by default
```

Comment:
Is this for files that are too big or patches that are too big?

Reply:
I'm going to assume it's the latter. It is identical to when GitHub hides files/doesn't load them because the diff is too large. I'll update the text.